src/transformers/integrations/deepspeed.py

```diff
@@ -289,6 +290,19 @@ def is_deepspeed_zero3_enabled():
         return False
 
 
+def is_deepspeed_zero3_enabled_accelerate():
```

Comment:
I don't think this check is necessary. I rechecked the code and creating an instance of `TrainingArguments` is necessary for `zero_init`.

Reply:
yeah, but if TrainingArguments is init after the model, it just fails silently.

Reply:
To put it differently, I don't think this is the correct way to check that `TrainingArguments` is already created. This checks env variables of accelerate which can be available even without `TrainingArguments` being created as it can set via `accelerate launch` command.